src/atc/delta/delta_handle.py

```diff
@@ -0,0 +1,95 @@
+from pyspark.sql import DataFrame
+
+from atc.atc_exceptions import AtcException
+from atc.config_master import TableConfigurator
+from atc.functions import init_dbutils
+from atc.spark import Spark
+
+
+class DeltaHandleException(AtcException):
+    pass
+
+
+class DeltaHandleInvalidName(DeltaHandleException):
+    pass
+
+
+class DeltaHandleInvalidFormat(DeltaHandleException):
+    pass
+
+
+class DeltaHandle:
+    def __init__(self, name: str, location: str = None, data_format: str = "delta"):
+        self._name = name
+        self._location = location
+        self._data_format = data_format
+
+        self._validate()
+
+    @classmethod
+    def from_tc(cls, id: str):
+        tc = TableConfigurator()
+        return cls(
+            name=tc.table_name(id),
+            location=tc.table_path(id),
+            data_format=tc.table_property(id, "format", "delta"),
+        )
+
+    def _validate(self):
+        # name is either `db`.`table` or just `table`
+        name_parts = self._name.split(".")
+        if len(name_parts) == 1:
+            self._db = None
+            self._table_name = name_parts[0]
+        elif len(name_parts) == 2:
+            self._db = name_parts[0]
+            self._table_name = name_parts[1]
+        else:
+            raise DeltaHandleInvalidName(f"Could not parse name {self._name}")
+
+        # only format delta is supported.
+        if self._data_format != "delta":
+            raise DeltaHandleInvalidFormat("Only format delta is supported.")
+
+    def read(self) -> DataFrame:
+        """Read table by path if location is given, otherwise from name."""
+        if self._location:
+            return Spark.get().read.format(self._data_format).load(self._location)
+        return Spark.get().table(self._name)
+
+    def write_or_append(self, df: DataFrame, mode: str, overwriteSchema=None) -> None:
```

Comment:
What is the datatype of "overwriteSchema"?

def write_or_append(self, df: DataFrame, mode: str, overwriteSchema: str ="false") -> None:

writer = df.write.format(self._data_format).mode(mode).option("overwriteSchema", overwriteSchema)

then remove line 62 and 63?

Reply:
i want this to be a tri-state bool True/False/None. Because any option set here overrides the cluster-wide settings which may not be what you want. Therefore if the value is None, I do not apply it.